DKP-312: Add release notes for 4.12.0

andrei-z-albu opened the pull request:
<!--Thanks for your contribution. See [CONTRIBUTING](CONTRIBUTING.md)
    for this project's contribution guidelines. Remove these comments
    as you go.-->

### Proposed changes

<!--Tell us what you did and why-->

### Related issues (optional)

<!--Refer to related PRs or issues: #1234, or 'Fixes #1234' or 'Closes #1234'.
    Or link to full URLs to issues or pull requests in other Github projects -->

thaJeztah commented on desktop/release-notes.md:
```diff
@@ -24,6 +24,75 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+
+
 ## Docker Desktop 4.11.1
 2022-08-05
```
We need to remove the `{% include desktop-install.html %}` for 4.11.1 and replace it with a block with regular links (and checksum section) similar as the 4.11.0 release, otherwise it shows the "install" block twice for the latest release.

https://github.com/docker/docker.github.io/blame/d282ce0a6360e70d88131d37f814ee63cb785d5d/desktop/release-notes.md#L49-L77

andrei-z-albu replied:
Thanks! Fixed with a new push

thaJeztah reviewed (changes requested):
Found an issue (left comment inline)

aevesdocker commented on desktop/release-notes.md:
```diff
@@ -24,6 +24,75 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+>
+> {% include desktop-install.html %}
+
+## New
```
I think some of these can be redistributed to the Bug fixes and minor changes sections. I'd suggest moving:

- Minimum version to install or update Docker Desktop on macOS is now 10.15
-Improved overall performance issues caused by calculating disk size. Related to [docker/for-win#9401](https://github.com/docker/for-win/issues/9401).
-  Enable ComposeV2 after factory reset
- Upgrade kernel to 5.10.124
- Prevents users on ARM macs without Rosetta installed from switching back to ComposeV1 (only has intel binaries).
- - Descending default sort order for volumes size and created column, containers started column
- Defined sort order for container/compose state column running > some running > paused > some paused > exited > some exited > created
- Re-organized container row actions
- - Runs every command on the Quickstart guide immediately.
- Updated the tooltip design.

andrei-z-albu replied:
RIght, thanks! I have moved these

aevesdocker commented:
@andrei-z-albu thank you for submitting the PR. Once the structural issues have been addressed (distributing some of the items from the 'New' section, and Sebastiaan's comment) then I can do a more thorough review to make sure it aligns with our Style Guide

mat007 commented on desktop/release-notes.md:
```diff
@@ -24,6 +24,75 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+
+## New
+
+- Minimum version to install or update Docker Desktop on macOS is now 10.15
```
I have just realized that there was a for-mac issue related to this:
```suggestion
  - Minimum version to install Docker Desktop on macOS is now 10.15. Fixes [docker/for-mac#6007](https://github.com/docker/for-mac/issues/6007)
```

mat007 commented on desktop/release-notes.md:
```diff
@@ -24,6 +24,75 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+- [mac] Quit Docker Desktop completely from the Dock menu (https://github.com/docker/for-mac/issues/6440)
+- Enable ComposeV2 after factory reset
+- Copied terminal contents do not contain non-breaking spaces anymore
+- Upgrade kernel to 5.10.124
```
I believe all entries should be in the past form:
```suggestion
- Upgraded kernel to 5.10.124
```
However this one should be in the `Upgrades` section as
```
[Linux kernel 5.10.104](https://hub.docker.com/layers/for-desktop-kernel/docker/for-desktop-kernel/5.10.124-78a158e1ce5b203564f7436fda87b288e257d506/images/sha256-2faf709b9fe0962e4a151df6b4b48d2cf5636619ebf970584c6559ef0c1df940?context=explore)
```

mat007 commented on desktop/release-notes.md:
```diff
@@ -24,6 +24,75 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+- Enable ComposeV2 after factory reset
+- Copied terminal contents do not contain non-breaking spaces anymore
+- Upgrade kernel to 5.10.124
+- Enables ComposeV2 by default on new installations/if there is no previous configuration.
```
```suggestion
- Enabled Compose V2 by default on new installations of Docker Desktop.
```

mat007 commented on desktop/release-notes.md:
```diff
@@ -24,6 +24,75 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+- Added View menu with Zoom In, Zoom Out, and Actual Size options to control the size of the Docker Desktop UI.
+- Descending default sort order for volumes size and created column, containers started column
+- Defined sort order for container/compose state column running > some running > paused > some paused > exited > some exited > created
+- Re-organized container row actions
```
I think that change doesn’t mean much for users, unless we can describe more in details what this is about?

andrei-z-albu replied:
Hmm, looks to me like most actions have been moved to the expand menu in the Action row, other than start/stop and delete. I don't know what 'pin action column' means from the [PR](https://github.com/docker/pinata/pull/18543). I have installed the new RC and this column doesn't seem pinned to me the way I imagine pinning should work. Do you think we should briefly describe the moved action items though?
Something like: Re-organized container row actions by keeping only the start/stop and delete actions visible at all times, while allowing access to the rest via the row menu item.

aevesdocker replied:
Your suggestion LGTM @andrei-z-albu 👍

mat007 commented on desktop/release-notes.md:
```diff
@@ -24,6 +24,75 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+- Runs every command on the Quickstart guide immediately.
+- Remove workaround for virtiofsd <-> qemu protocol mismatch on Fedora 35, as it is no longer needed. Fedora 35 users should upgrade the qemu package to the most recent version (qemu-6.1.0-15.fc35 as of the time of writing).
+- Implemented an integrated terminal for containers.
+- Updated the tooltip design.
```
This should either be removed or we should explain a bit more what it means?

andrei-z-albu replied:
Hmm, from what I can understand out of the PR, this styling change is done so we better adhere to our own UI guidelines. Perhaps a message such as this would be more relevant to users? (otherwise I agree we should just remove this entry):
Updates the tooltips in Docker Desktop and MUI theme package to align with the overall system design.

aevesdocker replied:
Like your suggestion @andrei-z-albu, would just change 'Updates' to 'Updated'

andrei-z-albu replied:
Great! I'll change it thus

mat007 commented on desktop/release-notes.md:
```diff
@@ -24,6 +24,75 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+- Remove workaround for virtiofsd <-> qemu protocol mismatch on Fedora 35, as it is no longer needed. Fedora 35 users should upgrade the qemu package to the most recent version (qemu-6.1.0-15.fc35 as of the time of writing).
+- Implemented an integrated terminal for containers.
+- Updated the tooltip design.
+- Show address for external links.
```
I have no idea what this means. 😅

andrei-z-albu replied:
Taken from the [PR](https://github.com/docker/pinata/pull/18547), the description there sounds like a better candidate for our release notes: "Adds a tooltip that displays the link address for all external links by default". What do you think?

mat007 commented on desktop/release-notes.md:
```diff
@@ -24,6 +24,75 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+### Upgrades
+
+- [Docker Compose v2.10.2](https://github.com/docker/compose/releases/tag/v2.10.2)
+- [Docker Compose "Cloud Integrations" v1.0.29](https://github.com/docker/compose-cli/releases/tag/v1.0.29)
```
I don’t think we’ve been listing that component in the release notes in the past, have we?

aevesdocker replied:
If it's not in previous release note patterns, I'd remove it

mat007 commented on desktop/release-notes.md:
```diff
@@ -24,6 +24,75 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+- [Go 1.19](https://github.com/golang/go/releases/tag/go1.19)
+- [cri-dockerd v0.2.5](https://github.com/Mirantis/cri-dockerd/releases/tag/v0.2.5)
+- [Buildx v0.9.1](https://github.com/docker/buildx/releases/tag/v0.9.1)
+- [containerd v1.6.8](https://github.com/containerd/containerd/releases/tag/v1.6.8)
+- [containerd v1.6.7](https://github.com/containerd/containerd/releases/tag/v1.6.7)
```
Only one is probably enough?
```suggestion
- [containerd v1.6.8](https://github.com/containerd/containerd/releases/tag/v1.6.8)
```

andrei-z-albu replied:
Feels to me that one bullet with two links not would be as readable as two separate bullets, assuming we want to showcase both versions in the release notes. But I don't have a strong opinion about it

andrei-z-albu replied:
What do you think @thaJeztah and @mat007 , shall we leave both entries in?

mat007 commented on desktop/release-notes.md:
```diff
@@ -24,6 +24,75 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+- [Buildx v0.9.1](https://github.com/docker/buildx/releases/tag/v0.9.1)
+- [containerd v1.6.8](https://github.com/containerd/containerd/releases/tag/v1.6.8)
+- [containerd v1.6.7](https://github.com/containerd/containerd/releases/tag/v1.6.7)
+- [runc v1.1.4](https://github.com/opencontainers/runc/releases/tag/v1.1.4)
+- [runc v1.1.3](https://github.com/opencontainers/runc/releases/tag/v1.1.3)
```
Same here
```suggestion
- [runc v1.1.4](https://github.com/opencontainers/runc/releases/tag/v1.1.4)
```

thaJeztah replied:
See above

mat007 commented on desktop/release-notes.md:
```diff
@@ -24,6 +24,75 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+- Fixed issues with the image list appearing empty in Docker Desktop even though there are images. Related to [docker/for-win#12693](https://github.com/docker/for-win/issues/12693) and [docker/for-mac#6347](https://github.com/docker/for-mac/issues/6347).
+- Fixed a bug that made Docker clients in some languages hang on `docker exec`. Fixes [https://github.com/apocas/dockerode/issues/534](https://github.com/apocas/dockerode/issues/534).
+- Extensions SDK: prevent failed spawned command to make Docker Desktop unexpectedly quit.
+- Fixes container terminal resizing issues.
```
That’s odd because further up we have an entry saying we have just added that as a new feature.
We don’t need to document bug fixes for not yet released features. 😄

andrei-z-albu replied:
Oh, right, that is odd to leave in there. Will take it out :D

mat007 commented on desktop/release-notes.md:
```diff
@@ -24,6 +24,75 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+- Fixed a bug that made Docker clients in some languages hang on `docker exec`. Fixes [https://github.com/apocas/dockerode/issues/534](https://github.com/apocas/dockerode/issues/534).
+- Extensions SDK: prevent failed spawned command to make Docker Desktop unexpectedly quit.
+- Fixes container terminal resizing issues.
+- Fixes extensions being displayed as disabled in the left menu but they are not.
```
```suggestion
- Fixed extensions being displayed as disabled in the left menu when they are not.
```

mat007 commented on desktop/release-notes.md:
```diff
@@ -24,6 +24,75 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+- Extensions SDK: prevent failed spawned command to make Docker Desktop unexpectedly quit.
+- Fixes container terminal resizing issues.
+- Fixes extensions being displayed as disabled in the left menu but they are not.
+- Fix `docker login` to private registries when Registry Access Management is enabled and access to Docker Hub is blocked.
```
```suggestion
- Fixed `docker login` to private registries when Registry Access Management is enabled and access to Docker Hub is blocked.
```

mat007 commented on desktop/release-notes.md:
```diff
@@ -24,6 +24,75 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+- Minimum version to install or update Docker Desktop on macOS is now 10.15
+- Improved overall performance issues caused by calculating disk size. Related to [docker/for-win#9401](https://github.com/docker/for-win/issues/9401).
+- Added the experimental docker daemon with the containerd image store.
+- Allow Docker Desktop to run untagged image (https://github.com/docker/for-mac/issues/6425)
```
```suggestion
- Allow Docker Desktop to run untagged images (https://github.com/docker/for-mac/issues/6425)
```
?

mat007 commented on desktop/release-notes.md:
```diff
@@ -24,6 +24,75 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+- Added the experimental docker daemon with the containerd image store.
+- Allow Docker Desktop to run untagged image (https://github.com/docker/for-mac/issues/6425)
+- Add search capabilities to Docker Extension Marketplace. Fixes [docker/roadmap#346](https://github.com/docker/roadmap/issues/346).
+- [mac] Quit Docker Desktop completely from the Dock menu (https://github.com/docker/for-mac/issues/6440)
```
That looks like a bug fix for the Mac section below?
Also the phrasing is a little odd…

andrei-z-albu replied:
You're right, I'll move it to the Mac section. Not sure about the phrasing though, perhaps make it much more explicit like:Quit Docker Desktop completely from the Dock menu, to ensure Docker Desktop isn't running in the background without the whale icon in the menu bar visible. Fixes [https://github.com/docker/for-mac/issues/6440]

mat007 reviewed (commented):
I haven’t checked all entries, but I think we should first move most of the New ones to the proper sections, like @aevesdocker suggested, before we can have a more thorough look.

andrei-z-albu commented:
> 

@aevesdocker I have moved the entries you have mentioned to the relevant section, and updated the 4.11.1 part with the relevant installer links and checksums, as per @thaJeztahs' comment. There are still a few unresolved questions, perhaps a more thorough review of yours could shed some light on these, so we push out clear and consistent release notes

lizzthabet commented on desktop/release-notes.md:
```diff
@@ -24,12 +24,118 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+- Re-organized container row actions.
+- Run every command on the Quickstart guide immediately.
+- Defined sort order for container/compose state column running > some running > paused > some paused > exited > some exited > created.
+- Fixed issues with the image list appearing empty in Docker Desktop even though there are images. Related to [docker/for-win#12693](https://github.com/docker/for-win/issues/12693) and [docker/for-mac#6347](https://github.com/docker/for-mac/issues/6347).
```
with this bug fix, i didn't note another change in behavior that could be noticeable to users. would it be possible to add it?
```
Defined what images are "in use" based on whether or not system containers are displayed; if system containers related to Kubernetes and Extensions are not displayed, the related images will not be defined as "in use."
```
(it's a difficult change to articulate succinctly, so feel free to modify if my phrasing is confusing!)

aevesdocker replied:
Small tweak to your suggestion:
Defined what images are "in use" based on whether or not system containers are displayed. If system containers related to Kubernetes and Extensions are not displayed, the related images are not defined as "in use."

andrei-z-albu replied:
Added your revised description @aevesdocker

lizzthabet replied:
thank you both!

aevesdocker commented on desktop/release-notes.md:
```diff
@@ -24,12 +24,118 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+- Allow Docker Desktop to run untagged images. Fixes [https://github.com/docker/for-mac/issues/6425]
+- Add search capabilities to Docker Extension Marketplace. Fixes [docker/roadmap#346](https://github.com/docker/roadmap/issues/346).
+- Copied terminal contents do not contain non-breaking spaces anymore.
+- Enabled Compose V2 by default on new installations of Docker Desktop.
```
```suggestion
-  Compose V2 is now enabled by default on new installations of Docker Desktop.
```

aevesdocker commented on desktop/release-notes.md:
```diff
@@ -24,12 +24,118 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+- Add search capabilities to Docker Extension Marketplace. Fixes [docker/roadmap#346](https://github.com/docker/roadmap/issues/346).
+- Copied terminal contents do not contain non-breaking spaces anymore.
+- Enabled Compose V2 by default on new installations of Docker Desktop.
+- Added View menu with Zoom In, Zoom Out, and Actual Size options to control the size of the Docker Desktop UI.
```
Maybe i missed something, but i can't see this feature in DD 🤔

andrei-z-albu replied:
On mac there's a new View menu next to Docker Desktop and Edit at the top, with the available actions mentioned here. I assume this shows up on win and linux in a similar fashion: ![Screenshot 2022-08-31 at 23 08 02](https://user-images.githubusercontent.com/54431776/187773486-bfbaee1a-8520-4232-ad83-050d45a85269.png)

aevesdocker replied:
Ah ok, I have Windows so can't see this. Think this should be moved to the Mac section in 'Bugs and minor changes'

StefanScherer replied:
The Zoom In, Zoom Out, and Actual Size also works for Windows with the keyboard shortcuts CTRL +/CTRL -/CTRL 0. So the feature itself works for all, but Windows Docker Dashboard doesn't have any menu like on Mac.
Maybe we should rephrase it a bit, because it's still an useful feature on Windows as well.

aevesdocker replied:
How about:

Added the ability to zoom in, out or set Docker Desktop to Actual Size. This is through the **View** menu on Mac or using keyboard shortcuts `CTRL +`, `CTRL -`, `CTRL 0` on Windows

andrei-z-albu replied:
Minor suggestion to include shortcuts on mac as well: Added the ability to zoom in, out or set Docker Desktop to Actual Size. This is done by using keyboard shortcuts `⌘ +` / `CTRL +`, `⌘ -` / `CTRL -`, `⌘ 0` / `CTRL 0` on Mac and Windows respectively, or through the View menu on Mac.

aevesdocker replied:
Perfect 👍

aevesdocker commented on desktop/release-notes.md:
```diff
@@ -24,12 +24,118 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+- Enabled Compose V2 by default on new installations of Docker Desktop.
+- Added View menu with Zoom In, Zoom Out, and Actual Size options to control the size of the Docker Desktop UI.
+- Added compose stop button if any related container is stoppable.
+- Individual compose containers are now deletable from overview.
```
```suggestion
- Individual compose containers are now deletable from the **Container** view. 
```

aevesdocker commented on desktop/release-notes.md:
```diff
@@ -24,12 +24,118 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+- Added View menu with Zoom In, Zoom Out, and Actual Size options to control the size of the Docker Desktop UI.
+- Added compose stop button if any related container is stoppable.
+- Individual compose containers are now deletable from overview.
+- Remove workaround for virtiofsd <-> qemu protocol mismatch on Fedora 35, as it is no longer needed. Fedora 35 users should upgrade the qemu package to the most recent version (qemu-6.1.0-15.fc35 as of the time of writing).
```
```suggestion
- Removed the workaround for virtiofsd <-> qemu protocol mismatch on Fedora 35, as it is no longer needed. Fedora 35 users should upgrade the qemu package to the most recent version (qemu-6.1.0-15.fc35 as of the time of writing).
```

aevesdocker commented on desktop/release-notes.md:
```diff
@@ -24,12 +24,118 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+- Individual compose containers are now deletable from overview.
+- Remove workaround for virtiofsd <-> qemu protocol mismatch on Fedora 35, as it is no longer needed. Fedora 35 users should upgrade the qemu package to the most recent version (qemu-6.1.0-15.fc35 as of the time of writing).
+- Implemented an integrated terminal for containers.
+- Show address for external links.
```
```suggestion
- Added a tooltip that displays the link address for all external links by default.
```

aevesdocker commented on desktop/release-notes.md:
```diff
@@ -24,12 +24,118 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+
+#### For all platforms
+
+- Enable Compose V2 after factory reset.
```
```suggestion
- Compose V2 is now enabled after factory reset.
```

aevesdocker commented on desktop/release-notes.md:
```diff
@@ -24,12 +24,118 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+#### For all platforms
+
+- Enable Compose V2 after factory reset.
+- Upgraded kernel to 5.10.124
```
```suggestion
- Upgraded kernel to 5.10.124.
```

aevesdocker commented on desktop/release-notes.md:
```diff
@@ -24,12 +24,118 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+- Enable Compose V2 after factory reset.
+- Upgraded kernel to 5.10.124
+- Improved overall performance issues caused by calculating disk size. Related to [docker/for-win#9401](https://github.com/docker/for-win/issues/9401).
+- Prevent users on ARM macs without Rosetta installed from switching back to Compose V1, which has only intel binaries.
```
```suggestion
- Docker Desktop now prevents users on ARM macs without Rosetta installed from switching back to Compose V1, which has only intel binaries.
```

aevesdocker commented on desktop/release-notes.md:
```diff
@@ -24,12 +24,118 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+- Upgraded kernel to 5.10.124
+- Improved overall performance issues caused by calculating disk size. Related to [docker/for-win#9401](https://github.com/docker/for-win/issues/9401).
+- Prevent users on ARM macs without Rosetta installed from switching back to Compose V1, which has only intel binaries.
+- Descending default sort order for volumes size and created column, containers started column.
```
```suggestion
- Changed the default sort order to descending for volume size and the **Created** column, along with the container's **Started** column.
```

aevesdocker commented on desktop/release-notes.md:
```diff
@@ -24,12 +24,118 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+- Prevent users on ARM macs without Rosetta installed from switching back to Compose V1, which has only intel binaries.
+- Descending default sort order for volumes size and created column, containers started column.
+- Re-organized container row actions.
+- Run every command on the Quickstart guide immediately.
```
```suggestion
- The Quickstart guide now runs every command immediately.
```

aevesdocker commented on desktop/release-notes.md:
```diff
@@ -24,12 +24,118 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+- Descending default sort order for volumes size and created column, containers started column.
+- Re-organized container row actions.
+- Run every command on the Quickstart guide immediately.
+- Defined sort order for container/compose state column running > some running > paused > some paused > exited > some exited > created.
```
```suggestion
- Defined the sort order for container/compose **Status** column to running > some running > paused > some paused > exited > some exited > created.
```

aevesdocker commented on desktop/release-notes.md:
```diff
@@ -24,12 +24,118 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+- Defined sort order for container/compose state column running > some running > paused > some paused > exited > some exited > created.
+- Fixed issues with the image list appearing empty in Docker Desktop even though there are images. Related to [docker/for-win#12693](https://github.com/docker/for-win/issues/12693) and [docker/for-mac#6347](https://github.com/docker/for-mac/issues/6347).
+- Fixed a bug that made Docker clients in some languages hang on `docker exec`. Fixes [https://github.com/apocas/dockerode/issues/534](https://github.com/apocas/dockerode/issues/534).
+- Extensions SDK: prevent failed spawned command to make Docker Desktop unexpectedly quit.
```
```suggestion
- A failed spawned command when building an extension no longer causes Docker Desktop to unexpectedly quit.
```

aevesdocker commented on desktop/release-notes.md:
```diff
@@ -24,12 +24,118 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+- Fixed issues with the image list appearing empty in Docker Desktop even though there are images. Related to [docker/for-win#12693](https://github.com/docker/for-win/issues/12693) and [docker/for-mac#6347](https://github.com/docker/for-mac/issues/6347).
+- Fixed a bug that made Docker clients in some languages hang on `docker exec`. Fixes [https://github.com/apocas/dockerode/issues/534](https://github.com/apocas/dockerode/issues/534).
+- Extensions SDK: prevent failed spawned command to make Docker Desktop unexpectedly quit.
+- Fixed extensions being displayed as disabled in the left menu when they are not.
```
```suggestion
- Fixed a bug that caused extensions to be displayed as disabled in the left menu when they are not.
```

aevesdocker commented on desktop/release-notes.md:
```diff
@@ -24,12 +24,118 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+- Extensions SDK: prevent failed spawned command to make Docker Desktop unexpectedly quit.
+- Fixed extensions being displayed as disabled in the left menu when they are not.
+- Fixed `docker login` to private registries when Registry Access Management is enabled and access to Docker Hub is blocked.
+- Fixed bug where we fail to start the Kubernetes cluster if the current cluster metadata is not stored in the `.kube/config` file.
```
```suggestion
- Fixed a bug where Docker Desktop fails to start the Kubernetes cluster if the current cluster metadata is not stored in the `.kube/config` file.
```

aevesdocker commented on desktop/release-notes.md:
```diff
@@ -24,12 +24,118 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+#### For Mac
+
+- Minimum version to install or update Docker Desktop on macOS is now 10.15. Fixes [docker/for-mac#6007](https://github.com/docker/for-mac/issues/6007).
+- Fixed a bug where Tray menu could incorrectly displays "Download will start soon..." after downloading the update. Fixes some issue reported in [for-mac/issues#5677](https://github.com/docker/for-mac/issues/5677)
```
```suggestion
- Fixed a bug where the Tray menu incorrectly displays "Download will start soon..." after downloading the update. Fixes some issue reported in [for-mac/issues#5677](https://github.com/docker/for-mac/issues/5677)
```

aevesdocker commented on desktop/release-notes.md:
```diff
@@ -24,12 +24,118 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+- Minimum version to install or update Docker Desktop on macOS is now 10.15. Fixes [docker/for-mac#6007](https://github.com/docker/for-mac/issues/6007).
+- Fixed a bug where Tray menu could incorrectly displays "Download will start soon..." after downloading the update. Fixes some issue reported in [for-mac/issues#5677](https://github.com/docker/for-mac/issues/5677)
+- Fixed a bug that didn't restart Docker Desktop after applying an update.
+- When using virtualization.framework, fix bugs where connection to Docker is lost when the computer sleeps and when using restrictive firewall software.
```
```suggestion
- Fixed a bug that caused the connection to Docker to be lost when the computer sleeps if a user is using virtualization.framework and restrictive firewall software.
```

aevesdocker commented on desktop/release-notes.md:
```diff
@@ -24,12 +24,118 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+- Fixed a bug where Tray menu could incorrectly displays "Download will start soon..." after downloading the update. Fixes some issue reported in [for-mac/issues#5677](https://github.com/docker/for-mac/issues/5677)
+- Fixed a bug that didn't restart Docker Desktop after applying an update.
+- When using virtualization.framework, fix bugs where connection to Docker is lost when the computer sleeps and when using restrictive firewall software.
+- Quit Docker Desktop completely from the Dock menu, to ensure Docker Desktop isn't running in the background without the whale icon in the menu bar visible. Fixes [https://github.com/docker/for-mac/issues/6440]
```
```suggestion
- Fixed a bug that caused Docker Desktop to run in the background even after a user had quit the application.  Fixes [https://github.com/docker/for-mac/issues/6440]
```

aevesdocker commented on desktop/release-notes.md:
```diff
@@ -24,12 +24,118 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+### Security 
+
+#### For Windows
+- Fix a bypass for the `--no-windows-containers` installation flag which was introduced in version 4.11. This flag allows administrators to disable the use of Windows containers.
```
```suggestion
- Fixed a bypass for the `--no-windows-containers` installation flag which was introduced in version 4.11. This flag allows administrators to disable the use of Windows containers.
```

aevesdocker commented on desktop/release-notes.md:
```diff
@@ -24,12 +24,118 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+
+#### For Windows
+- Fix a bypass for the `--no-windows-containers` installation flag which was introduced in version 4.11. This flag allows administrators to disable the use of Windows containers.
+- Fix argument injection to the Docker Desktop installer which may result in local privilege escalation.
```
```suggestion
- Fixed the argument injection to the Docker Desktop installer which may result in local privilege escalation.
```

aevesdocker commented on desktop/release-notes.md:
```diff
@@ -24,12 +24,118 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+- Fix a bypass for the `--no-windows-containers` installation flag which was introduced in version 4.11. This flag allows administrators to disable the use of Windows containers.
+- Fix argument injection to the Docker Desktop installer which may result in local privilege escalation.
+
+#### For all platforms
```
Can we put this section above the "For Windows" section?

andrei-z-albu commented on desktop/release-notes.md:
```diff
@@ -24,12 +24,118 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+- Added the experimental docker daemon with the containerd image store.
+- Allow Docker Desktop to run untagged images. Fixes [https://github.com/docker/for-mac/issues/6425]
+- Add search capabilities to Docker Extension Marketplace. Fixes [docker/roadmap#346](https://github.com/docker/roadmap/issues/346).
+- Copied terminal contents do not contain non-breaking spaces anymore.
```
Will move it

hyu commented on desktop/release-notes.md:
```diff
@@ -24,12 +24,118 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+- Added the ability to use containerd for pulling and storing images. This is an experimental feature. 
+- Docker Desktop now runs untagged images. Fixes [https://github.com/docker/for-mac/issues/6425]
+- Added search capabilities to Docker Extension's Marketplace. Fixes [docker/roadmap#346](https://github.com/docker/roadmap/issues/346).
+-  Compose V2 is now enabled by default on new installations of Docker Desktop.
```
```suggestion
- Compose V2 is now enabled by default on new installations of Docker Desktop.
- Precedence order of environment variables in Compose is more consistent, and clearly [documented](https://docs.docker.com/compose/envvars-precedence/).
```

Sorry for the late addition to the release notes! The updated version of Compose included in this release will include some changes to the implementation of environment variables. 

In the transition from v1 to v2, we received some user feedback that existing env var usage wasn't resulting in expected behavior. Since the evolution of the env vars implementation in v1 was quite organic, the rewrite in Go for v2 resulted in subtle changes. Instead of replicating the exact edge cases from v1, we address this by clearly documenting the precedence order in the latest version of Compose. 

It's debatable whether this is actually "New," but I placed the item here for now since we want users to be aware of this change. If you think it's more appropriate for another section, feel free to move it.

aevesdocker replied:
Thanks for the addition Han. Wording LGTM but think this should be moved to the 'Bugs and minor changes' section @andrei-z-albu

aevesdocker commented on desktop/release-notes.md:
```diff
@@ -24,12 +24,118 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+- Fixed a bug that caused extensions to be displayed as disabled in the left menu when they are not.
+- Fixed `docker login` to private registries when Registry Access Management is enabled and access to Docker Hub is blocked.
+- Fixed a bug where Docker Desktop fails to start the Kubernetes cluster if the current cluster metadata is not stored in the `.kube/config` file.
+- UUpdated the tooltips in Docker Desktop and MUI theme package to align with the overall system design.
```
```suggestion
- Updated the tooltips in Docker Desktop and MUI theme package to align with the overall system design.
```

aevesdocker commented on desktop/release-notes.md:
```diff
@@ -24,12 +24,119 @@ Take a look at the [Docker Public Roadmap](https://github.com/docker/roadmap/pro
+### New
+
+- Added the ability to use containerd for pulling and storing images. This is an experimental feature. 
+- Docker Desktop now runs untagged images. Fixes [https://github.com/docker/for-mac/issues/6425]
```
```suggestion
- Docker Desktop now runs untagged images. Fixes [docker/for-mac#6425](https://github.com/docker/for-mac/issues/6425).
```

aevesdocker reviewed (approved):
Thank you @andrei-z-albu! This LGTM

StefanScherer commented on desktop/release-notes.md:
Are the checksums up to date? I only verified this one on build 85629 and I got a different SHA-256 sum. 🤔

StefanScherer reviewed (approved):
LGTM
